query/text_parser.js

```diff
@@ -91,4 +102,11 @@ function shouldSetQueryIntoHouseNumber(vs) {
           /^[0-9]+$/.test(vs.var('input:query').toString());
 }
 
+function shouldTreatIslandAsCounty(vs) {
+  return vs.isset('input:island') &&
+    vs.isset('input:region') &&
+    !vs.isset('input:county') &&
+    vs.var('input:region').toString() === 'hawaii';
```

Comment:
this definitely calls for an awesome comment about why it's only `hawaii` at this time and the implications of this one-off fix. also is it safe to assume this will always be all lowercase?